flake.nix

```diff
@@ -59,16 +41,16 @@
 
               in
               map addModuleDeclarationFile modules ++ [
-                {
+                ({ config, ... }: {
                   system.nixos.versionSuffix =
                     ".${final.substring 0 8 (self.lastModifiedDate or self.lastModified or "19700101")}.${self.shortRev or "dirty"}";
                   system.nixos.revision = final.mkIf (self ? rev) self.rev;
 
                   system.build = {
-                    vm = vmConfig.system.build.vm;
-                    vmWithBootLoader = vmWithBootLoaderConfig.system.build.vm;
+                    vm = config.virtualisation.vmVariant.system.build.vm;
+                    vmWithBootLoader = config.virtualisation.vmVariantWithBootLoader.system.build.vm;
```

Comment:
Doesn't this cause infinite recursion?

Reply:
No, `extendModules` produces a new config.

I am concerned about this attr colliding though. What if someone invokes `nixosSystem` for the purpose of creating an image with the `qemu-vm.nix` module that they imported themselves.

Reply:
I've confirmed that it works, but I don't get why.

`extendModules` reuses the modules that were passed to `evalModules`, in this case one containing `system.build.vm = config.virtualisation.vmVariant.system.build.vm;`, so why doesn't `qemu-vm.nix` collide with that?

Reply:
`qemu-vm.nix` isn't in `module-list.nix`, so it's not part of the normal config unless you `imports` it manually.

It lines up almost too perfectly.

Reply:
But it is imported [here](https://github.com/NixOS/nixpkgs/pull/151082/files#diff-50176a86aa4300bb428500e2ff1d920bb67b8a216cf3e173e2896b5801645eaaR8-R10), I'd expect things to clash there

Reply:
It's because of legacy `types.attrs` :fearful:

Reply:
Ah... guess the `mkDefault` solution is fine